src/plugins/intel_cpu/src/nodes/interaction.cpp

```diff
@@ -0,0 +1,225 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <string>
+#include <vector>
+
+#include "ie_parallel.hpp"
+#include "ngraph_transformations/op/interaction.hpp"
+#include "interaction.h"
+#include "utils/general_utils.h"
+#include <onednn/dnnl.h>
+#include <dnnl_extension_utils.h>
+#include <immintrin.h>
+#include "nodes/common/cpu_convert.h"
+#include "memory_desc/cpu_memory_desc_utils.h"
+#include "memory_desc/dnnl_blocked_memory_desc.h"
+#include "common/bfloat16.hpp"
+#include "common/cpu_memcpy.h"
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+Interaction::Interaction(const std::shared_ptr<ngraph::Node>& op, const dnnl::engine& eng, WeightsSharing::Ptr &cache)
+        : Node(op, eng, cache) {
+    std::string errorMessage;
+    if (isSupportedOperation(op, errorMessage)) {
+        errorPrefix = "Interaction node with name '" + getName() + "'";
+    } else {
+        IE_THROW(NotImplemented) << errorMessage;
+    }
+}
+
+void Interaction::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+    dataPrecision = getOriginalInputPrecisionAtPort(0);
+    // if parent precision is not bf16/fp32, then set it to fp32. Current impl only support FP32 BF16
+    if (!one_of(dataPrecision, InferenceEngine::Precision::BF16, InferenceEngine::Precision::FP32))
+        dataPrecision = InferenceEngine::Precision::FP32;
```

Comment:
Why do you prefer FP32 here? Conversion will be executed anyway, but BF16 should be faster.

Reply:
In my thought, the inputs of **Interaction** is Embedding which only has FP32 output, therefore, in default case, the **Interaction** uses FP32 too. If I add the following at the beginning of `initSupportedPrimitiveDescriptors`, this means that the **Interaction** will run in BF16 no matter what user specifies. I think that breaks what the user expects.
```
void Interaction::initSupportedPrimitiveDescriptors() {
    if (!supportedPrimitiveDescriptors.empty())
        return;
    dataPrecision = getOriginalInputPrecisionAtPort(0);
  if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16)) {
    dataPrecision = InferenceEngine::Precision::BF16;
  }
```
For example, I follow the assumptions below.

1. If user sets `infer_precision=FP32`, all operations should run in FP32.
2. If user sets `infer_precision=BF16`, operation which supports BF16 should run in BF16
3. If user runs int8 model, then Interaction runs in FP32 just like IPEX framework.

I think the gap here is whether interaction should run in FP32 or BF16 in int8 models. What do you think ?